packages/collector/Collector_Helpers.php

```diff
@@ -0,0 +1,28 @@
+<?php
+function collector_get_tmpfile($name, $type)
+{
+    $tmpName = tempnam('/tmp/', 'clctr-'. date('Y-m-d_H-i-s-') . $name . '-');
```

Comment:
Let's avoid writing potentially sensitive data to `/tmp` and store everything in memory instead

Reply:
Looks like there is no way to add a file handle pointing to `php://memory` to a `ZipArchive`.

`ZipArchives` don't expose any methods that allow a filehandle to be added directly:

https://www.php.net/manual/en/class.ziparchive.php#ziparchive.synopsis

We can't re-open `php://memory` since each time we do that, it references a different "file":

https://php.net/manual/en/wrappers.php.php#example-1264 (example \#5)

My workaround is to use the wordpress-recommended `get_temp_dir()`:

https://developer.wordpress.org/reference/functions/get_temp_dir/

Reply:
How about reading from that in-memory file handle and adding it to the archive [addFromString](https://www.php.net/manual/en/ziparchive.addfromstring.php)?

Reply:
@adamziel How can we guarantee that we won't hit the memory limit in that case?

Reply:
@seanmorris We can't, but maybe that's fine. We can't guarantee we won't run out of disk space, either. Let's find out what are the actual constraints and then iterate on the approach. Perhaps folks will be hesitant even about this memory-based approach and we'll have to reinvent the entire idea. Or perhaps, later on, we won't actually need to ship all the files in that zip? In a later version, the media files could be downloaded separately. Similarly, some plugins could eventually be transmitted as plugin directory URLs for Playground to download separately. And so on.

Reply:
Switched to streaming in 22b6cd46f73f793a980c37a4801523cb686371fd